### PR TITLE
Category

### PR DESCRIPTION
**Pull request by SJTGSHIVAM**

### CODE REVIEW

File: src/components/CartCard.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      toastError();
+    }
+  };
+  const wishlistToggle = async (
+    id: string,
+    encodedToken: string,
+    invalidate: () => void
+  ) => {
+    try {
+      postUserWishlist(id, { authorinzation: encodedToken });
```
Comment (tejas-muthya37) [inline src/components/CartCard.tsx]: Typo here. *authorization.

File: src/components/Navbar.tsx [hunk 2]
```diff
@@ -7,11 +12,41 @@ import {
...
+  }, []);
+  const toggleTheme = () => {
+    if (globalThemeDark) {
+      setTheme({ theme: lightTheme });
+      localStorage.setItem("lg2__theme", "light");
+      setGlobalThemeDark(false);
+    } else {
+      setTheme();
+      localStorage.setItem("lg2__theme", "dark");
+      setGlobalThemeDark(true);
```
Comment (tejas-muthya37) [inline src/components/Navbar.tsx:45]: This step can be abstracted as setGlobalThemeDark(! globalThemeDark)

File: src/components/ProductCard.tsx [hunk 3]
```diff
@@ -62,13 +67,15 @@ export const ProductCard = ({
         await deleteUserWishlistProduct(id, { authorinzation: encodedToken });
```
Comment (tejas-muthya37) [inline src/components/ProductCard.tsx]: *authorization.

File: src/components/ProductCard.tsx [hunk 4]
```diff
@@ -62,13 +67,15 @@ export const ProductCard = ({
...
       } catch (error) {
         setIsInWishlistLocalState(true);
+        toastError();
       }
     } else {
       setIsInWishlistLocalState(true);
       try {
         await postUserWishlist(id, { authorinzation: encodedToken });
```
Comment (tejas-muthya37) [inline src/components/ProductCard.tsx]: *authorization.

File: src/components/WishlistCard.tsx [hunk 5]
```diff
@@ -45,8 +49,9 @@ export const WishlistCard = ({
   const addToCart = async (id: string, encodedToken: string) => {
     try {
       await postUserCart(id, { authorinzation: encodedToken });
```
Comment (tejas-muthya37) [inline src/components/WishlistCard.tsx]: Ctrl + F this word and replace it everywhere.

File: src/pages/Cart/Cart.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+import { CartProduct } from 'interfaces';
+import { Link } from 'react-router-dom';
+
+export const Cart = () => {
+  const [userCart, setUserCart] = useState<Array<CartProduct>>([]);
+  const [invalidateToggle, setInvalidateToggle] = useState(true);
+  const [price, setPrice] = useState(0);
+  const [discount, setDiscount] = useState(0);
+  const [total, setTotal] = useState(0);
+  const [deliveryCharges, setDeliveryCharges] = useState(0);
```
Comment (tejas-muthya37) [inline src/pages/Cart/Cart.tsx:20]: If possible, you can fit all of these into 1 reducer.

**Review by tejas-muthya37 — APPROVED** [review]
No issues as such, 3 things that I liked about your code,

1. Separation of concerns.
2. Grouping all APIs in a single file.
3. Level of detailing in styling.

File: src/components/Navbar.tsx (hunk 2 above)
Comment (SJTGSHIVAM) [reply src/components/Navbar.tsx:45]: No, I do not intend to toggle it here.

**Comment by SJTGSHIVAM** [comment]
tried to fix that Typo and something got messed up.

**Comment by SJTGSHIVAM** [comment]
@tejas-muthya37  and everyone else it was an issue with netlify cache I cleared it and redeployed it got fixed.

File: src/pages/Cart/Cart.tsx (hunk 6 above)
Comment (SJTGSHIVAM) [reply src/pages/Cart/Cart.tsx:20]: Yeah I am also thinking about it but I am not completely sure about it. There is only one action I can take as of now and hence I am not much willing. I need to check the comparison and rethink that. May be I'll take this up in a later sprint.
Once this submission is done let's plan a short call where we can discuss it. I would love to hear your opinion on that.

File: src/components/WishlistCard.tsx (hunk 5 above)
Comment (SJTGSHIVAM) [reply src/components/WishlistCard.tsx]: Yeah just did it

**Review by Hnh013 — APPROVED** [review]
checked the preview , the ui theme is good , a suggestion to use reducer instead of multiple useState

**Comment by SJTGSHIVAM** [comment]
> checked the preview , the ui theme is good , a suggestion to use reducer instead of multiple useState
As mentioned I will take it up in a later sprint after giving it some thought

**Pull request merged**